dds/DCPS/RTPS/RtpsSubmessageKind.idl

```diff
@@ -0,0 +1,40 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_RTPS_RTPSSUBMESSAGEKIND_IDL
+#define OPENDDS_DCPS_RTPS_RTPSSUBMESSAGEKIND_IDL
+
+module OpenDDS {
+module RTPS {
+
+enum SubmessageKind {
+  @value(0x00) RTPS_HE, /* HeaderExtension */
+  @value(0x01) PAD, /* Pad */
+  @value(0x06) ACKNACK, /* AckNack */
+  @value(0x07) HEARTBEAT, /* Heartbeat */
+  @value(0x08) GAP, /* Gap */
+  @value(0x09) INFO_TS, /* InfoTimestamp */
+  @value(0x0c) INFO_SRC, /* InfoSource */
+  @value(0x0d) INFO_REPLY_IP4, /* InfoReplyIp4 */
+  @value(0x0e) INFO_DST, /* InfoDestination */
+  @value(0x0f) INFO_REPLY, /* InfoReply */
+  @value(0x12) NACK_FRAG, /* NackFrag */
+  @value(0x13) HEARTBEAT_FRAG, /* HeartbeatFrag */
+  @value(0x15) DATA, /* Data */
+  @value(0x16) DATA_FRAG, /* DataFrag */
+
+  // SubmessageKinds 0x30-3f are reserved for DDS Security
+  @value(0x30) SEC_BODY,
+  @value(0x31) SEC_PREFIX,
+  @value(0x32) SEC_POSTFIX,
+  @value(0x33) SRTPS_PREFIX,
+  @value(0x34) SRTPS_POSTFIX
+
+  // SubmessageKinds 0x80 and above are vendor-specific
+};
```

Comment:
Since this uses the IDL-to-C++11 mappings, does it only work with compilers with C++11 support?

Reply:
It actually looks like he's compiling this file using `-Lspcpp`, which is the mapping for safety profile. Both use `langmap_generator`, so I'm guessing `@value` works for both C++11 and Safety profile mappings now and there's nothing stopping us from using the safety profile mapping like this in a normal build.

Reply:
Lspcpp is a minimal generator backend that's not useful for end users building their IDL for use as topic types.  It has enough of the IDL-to-C++(11) mapping to generate the code that's needed for safety profile when tao_idl isn't available.
In this case, we have IDL files in the OpenDDS_Rtps library that are just used for implementing the protocol and not as DDS topic types.